### DISCUSSION TITLE
Re-render <head> content differences when using client-side navigation?

### DISCUSSION THREAD

**Discussion by Spekachu** [WordPress/gutenberg#60512] [Interactivity API]
I know we can use data-wp-router-region on our interactive blocks so they will re-render their content on client-side navigation. However if that new content is using a stylesheet in the <head> which only loads on initial page load, that content won't display properly. Is there a solution for this? 
The new stylesheets are included in the fetch response when navigating, I just can't figure out how to apply them to the navigated page.

**Comment by DAreRodz** [comment]
Hi @Spekachu. Updating the `<head>` content is something we have in mind―especially when dealing with new assets (scripts, stylesheets). We have a preliminary solution that will be shipped with the [experimental full client-side navigation](https://github.com/WordPress/gutenberg/pull/59707) in Gutenberg and will be the same for the region-based navigation.

Still, this is something we need some time to design properly until we eventually release this feature. 🙂

I'm curious―what is your specific case? Do you have blocks that can appear conditionally? For now, as a workaround, maybe you could always inject the stylesheet. 🤔

**Reply by Spekachu** [reply]
I guess my use-case is full client-side navigation. I actually was able to get the new styles/scripts by fetching the new page, merging the new head with the old (current) one, and passing the html to the router navigate function so it doesn't fetch the new page again. 
I want to build an interactive shop. I'm using Woocommerce blocks on the category pages to show those specific products. 
Here's my progress so far using all interactive blocks:

https://github.com/WordPress/gutenberg/assets/13248014/5de429de-896e-4d1a-b809-215a120e1053

**Reply by luisherranz** [reply]
I love it! Do you have a link to see the website?
